___

# Experimento dos resistores - APS7 (um dos exercícios)

** Compraria o lote de resistores pelo valor médio e pela tolerância? **

Use a sua amostra para concluir se compraria o lote de resistores pelo controle de qualidade tanto quanto aos $1000 k\Omega$ e quanto aos $5\%$ de tolerância.

Use 95% de confiança para tomada de decisão!

In [12]:
import pandas as pd
import numpy as np

In [18]:
lista_Resistores = [1.035, 1.026,1.031,1.004,1.027,1.025,0.987,1.048,1.055,1.044,1.019,1.048,1.057]
lista_Resistores = pd.Series(lista_Resistores)
quantidade = len(lista_Resistores)

média = np.mean(lista_Resistores)
desvio = np.std(lista_Resistores,ddof=1)

print("Média: ", média)
print("Desvio padrão amostral: ", desvio)

Média:  1.0312307692307692
Desvio padrão amostral:  0.0202078938625225


In [19]:
média_bootstrap = []

rodagens = 15000
for i in range(rodagens):
    média = np.mean(np.random.choice(lista_Resistores, quantidade, replace=True))
    média_bootstrap.append(média)
média_bootstrap = pd.Series(média_bootstrap)

In [23]:
pontoem_doismeio = np.percentile(média_bootstrap, 2.5)
pontoem_noventaoitomeio = np.percentile(média_bootstrap, 100-2.5)
print(pontoem_doismeio,pontoem_noventaoitomeio)

1.0204615384615385 1.0412307692307694


Os resistores para o intervalo ed confianca de 95% apresentam valores entre 1.02 e 1.04, o que representa um erro de ate 4% do valor divulgado de resistência, portanto o lote pode ser comprado, uma vez que esta dentro da tolerância exijida pelo controle de qualidade.